src/orion/core/worker/trial.py

```diff
@@ -106,6 +106,10 @@ def to_dict(self):
                 )
             return ret
 
+        def __eq__(self, other):
```

Comment:
This should have been done earlier. It is useful fo the `hash_name` property as well. Shall I right a `__hash__` and change the implementation of the `hash_name`? I think I should. This will be in a future PR.

Reply:
Yes sure if there is hashing that is certainly related to __eq__.

Reply:
#89